distros/humble/rviz-ogre-vendor/rviz-ogre-vendor.patch

```diff
@@ -0,0 +1,55 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 3f5fda2ee..53b9c1af0 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -125,7 +125,11 @@ macro(build_ogre)
+     set(OGRE_CXX_FLAGS "${OGRE_CXX_FLAGS} /w /EHsc")
+   elseif(APPLE)
+     set(OGRE_CXX_FLAGS "${OGRE_CXX_FLAGS} -std=c++14 -stdlib=libc++ -w")
+-    list(APPEND extra_cmake_args "-DCMAKE_OSX_ARCHITECTURES='x86_64'")
++    if (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "arm64")
++      list(APPEND extra_cmake_args "-DCMAKE_OSX_ARCHITECTURES='arm64'")
++    else()
++      list(APPEND extra_cmake_args "-DCMAKE_OSX_ARCHITECTURES='x86_64'")
++    endif()
+   else()  # Linux
+     set(OGRE_C_FLAGS "${OGRE_C_FLAGS} -w")
+     # include Clang -Wno-everything to disable warnings in that build. GCC doesn't mind it
+@@ -161,9 +165,8 @@ macro(build_ogre)
+ 
+   find_package(Patch REQUIRED)
+   include(ExternalProject)
+-  ExternalProject_Add(ogre-v1.12.1
+-    URL https://github.com/OGRECave/ogre/archive/v1.12.1.zip
+-    URL_MD5 cdbea4006d223c173e0a93864111b936
++  ExternalProject_Add(ogre-v1.12.13
++    URL https://github.com/OGRECave/ogre/archive/v1.12.13.zip
+     TIMEOUT 1200
+     LOG_CONFIGURE ${should_log}
+     LOG_BUILD ${should_log}
+@@ -185,21 +188,21 @@ macro(build_ogre)
+       -DOGRE_BUILD_COMPONENT_JAVA:BOOL=FALSE
+       -DOGRE_BUILD_COMPONENT_CSHARP:BOOL=FALSE
+       -DOGRE_BUILD_COMPONENT_BITES:BOOL=FALSE
++      -DOGRE_ENABLE_PRECOMPILED_HEADERS=0
```

Comment:
This seems to be the actual needed change besides the version bump. If this is the case, can you mention which error is actually fixed by this in the commit message?

Reply:
Without this line we get `error: Objective-C was disabled in PCH file but is currently enabled`. Probably because the upstream don't provide precompiled headers for macOS.